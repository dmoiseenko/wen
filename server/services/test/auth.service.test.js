const authService = require('../auth.service');

const user = {
  id: 1,
  passwordHash: '123'
};
const email = 'email';
const password = 'password';


describe('getUserAndPassword', () => {
  it('should get user by email', async () => {
    const getUserByEmail = jest.fn().mockReturnValue(Promise.resolve(user));

    const actual = await authService.getUserAndPassword(getUserByEmail)({ email, password });

    expect(actual).toMatchSnapshot();
  });
});

describe('verifyPassword', () => {
  it('should verify password hash', async () => {
    const verifyPasswordHash = jest.fn().mockReturnValue(true);

    const actual = await authService.verifyPassword(verifyPasswordHash)({ user, password });

    expect(actual).toMatchSnapshot();
  });
});

describe('formPayload', () => {
  it('should form JWT token payload', () => {
    const actual = authService.formPayload({ user });

    expect(actual).toMatchSnapshot();
  });
});
