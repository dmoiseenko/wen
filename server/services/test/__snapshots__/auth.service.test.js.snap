// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formPayload should form JWT token payload 1`] = `
Object {
  "id": 1,
}
`;

exports[`getUserAndPassword should get user by email 1`] = `
Object {
  "password": "password",
  "user": Object {
    "id": 1,
    "passwordHash": "123",
  },
}
`;

exports[`verifyPassword should verify password hash 1`] = `
Object {
  "isVerified": true,
  "user": Object {
    "id": 1,
    "passwordHash": "123",
  },
}
`;
